src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
> …then that router has serious security flaw…

Security flaw is what this PR is going to merge: unlimited access for unknown and uncontrolled hosts for the sake of (imaginary?) convenience. IMO, you should not invent a wheel but implement this in the common way: give user a way to provide a list of sub-nets. Probably with a a button ("Add localhost and LAN addresses") for a convenience with a message like "This list is provided for convenience purposes only. Please remove all unneeded addresses"